Unrecognized characters generated during document construction

**Describe the bug**
file:zephyr\doc\_build\rst\doc\reference\devicetree\bindings.rst
```
W眉rth Elektronik GmbH. (we)
===========================

.. rst-class:: rst-columns

- :ref:`dtbinding_we_wsen_itds`

.. _dt_vendor_xlnx:
```
file:zephyr/dts/bindings/vendor-prefixes.txt
```
Würth Elektronik GmbH
```
Open this file in my system, the text is correct(utf-8)
The intermediate file generated by the build is incorrect，It seems that utf-8 is not used, generating


**Environment (please complete the following information):**
 -  Windows(Chinese)

Similarly, I can’t write Chinese characters in it, and I will report an error directly.
zephyr/dts/bindings/vendor-prefixes.txt
LOG:
```
FAILED: CMakeFiles/devicetree
cmd.exe /C "cd /D E:\work\zephyr\doc && "C:\Program Files\CMake\bin\cmake.exe" -E env PYTHONPATH=E:/work/zephyr/scripts/dts/python-devicetree/src; ZEPHYR_BASE=E:/work/ZC3827/ZCAN-IOT/zephyr GEN_DEVICETREE_REST_ZEPHYR_DOCSET= C:/Python39/python.exe E:/work/ZC3827/ZCAN-IOT/zephyr/doc/_scripts/gen_devicetree_rest.py --vendor-prefixes E:/work/ZC3827/ZCAN-IOT/zephyr/dts/bindings/vendor-prefixes.txt --dts-root E:/work/ZC3827/ZCAN-IOT/zephyr E:/work/ZC3827/ZCAN-IOT/zephyr/doc/_build/rst/doc/reference/devicetree"
ninja: build stopped: subcommand failed.
```

--- a/doc/_scripts/gen_devicetree_rest.py
+++ b/doc/_scripts/gen_devicetree_rest.py
@@ -78,7 +78,7 @@ class VndLookup:
             None: GENERIC_OR_VENDOR_INDEPENDENT,
         }
         found_separator = False     # have we found the '-----' separator?
-        with open(vendor_prefixes, 'r') as f:
+        with open(vendor_prefixes, 'r', encoding='utf-8') as f:
             for line in f:
                 line = line.strip()
                 if line and found_separator:

